cmd/uhc/cluster/list/list.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright (c) 2018 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package list
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/openshift-online/uhc-cli/pkg/config"
+	"github.com/openshift-online/uhc-cli/pkg/util"
+	"github.com/openshift-online/uhc-sdk-go/pkg/client"
+	"github.com/openshift-online/uhc-sdk-go/pkg/client/clustersmgmt/v1"
+	"github.com/spf13/cobra"
+)
+
+var args struct {
+	debug     bool
+	parameter []string
+	header    []string
+}
+
+var managed string
+
+var Cmd = &cobra.Command{
+	Use:   "list [MANAGED | UNMANAGED] ",
+	Short: "List clusters",
+	Long:  "List clusters. Defaults to managed.",
+	Run:   run,
+}
+
+func init() {
+	flags := Cmd.Flags()
+	flags.BoolVar(
+		&args.debug,
+		"debug",
+		false,
+		"Enable debug mode.",
+	)
+	flags.StringArrayVar(
+		&args.parameter,
+		"parameter",
+		nil,
+		"Query parameters to add to the request. The value must be the name of the "+
+			"parameter, followed by an optional equals sign and then the value "+
+			"of the parameter. Can be used multiple times to specify multiple "+
+			"parameters or multiple values for the same parameter.",
+	)
+	flags.StringArrayVar(
+		&args.header,
+		"header",
+		nil,
+		"Headers to add to the request. The value must be the name of the header "+
+			"followed by an optional equals sign and then the value of the "+
+			"header. Can be used multiple times to specify multiple headers "+
+			"or multiple values for the same header.",
+	)
+}
+func run(cmd *cobra.Command, argv []string) {
+
+	if len(argv) > 0 {
+		switch argv[0] {
+		case "managed":
+			managed = "managed='true'"
+		case "unmanaged":
+			managed = "managed='false'"
+		default:
+			fmt.Fprintf(os.Stderr, "Unrecognized filter\n")
+			os.Exit(1)
+		}
+	} else {
+		managed = "managed='true'"
+	}
```

Comment:
For this I'd suggest to use a flag like `--managed=true|false`. Or a more generic `--search` option, so that the user can do `--search="managed = 'true'"` or any other search that she wants to do.

Reply:
At least from our teams perspective, we really care about `managed` clusters. So to list all clusters that are managed by default is useful to us. This is really the goal of creating the `cluster` subcommand in general; to have a have a simple wrapper around the most common UHC api calls we need.
So these are the three most common use cases I can see:
1) List of managed clusters
2) List of unmanaged clusters
3) List of clusters with some parameter (i.e. through --parameter).

So IMO, typing `uhc cluster list managed` to get a list of managed clusters is much simpler and more helpful than `uhc cluster list --managed=true` or `uhc cluster list --search="managed='true'"`

Reply:
I understand technically we shouldn't need to add this but it's a significant support convenience. For some additional context, with ~50k clusters, only 26 are managed right now. The managed support team will be heavy users of UHC CLI. They would need to use the flag every single time.

Reply:
@aweiteka @fahlmant is the discussion about `managed` versus `--managed`? It may be more convenient to type a couiple less characters, but I think it doesn't justify breaking the general idea of using `--` for arguments that modify the behaviour of a command.

If the discussion is about not adding at all the `managed` or `--managed` option and instead always listing only managed cluster by default then I object. It is important that the tool is convenient for you and it is also important that is convenient for other users, including customers. For me, for example, it is very convenient to be able to get all the clusters, managed clusters aren't enough for me.

My suggestion is make it `--managed`, with an opitional `true|false` value. The default can be `true`, so for your use case you can just type `uhc cluster list --managed`, only two additional characters.

Reply:
@jhernand fixed to use `--managed`